plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastoreClient.java

```diff
@@ -762,7 +762,11 @@ public void createFunction(Function function)
     public void alterFunction(Function function)
             throws TException
     {
-        client.alterFunction(prependCatalogToDbName(catalogName, function.getDbName()), function.getFunctionName(), function);
+        // Hive 3 does not actually replace the content of the function
+        // https://github.com/apache/hive/blob/rel/release-3.1.2/standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java#L9310
+        // Fall back to use drop & create for doing the replace function operation
+        dropFunction(function.getDbName(), function.getFunctionName());
```

Comment:
Another alternative would be to  throw an `NOT_SUPPORTED` TrinoException in `io.trino.plugin.hive.metastore.thrift.BridgingHiveMetastore#replaceFunction` but this would make the `CREATE OR REPLACE` a bit cumbersome - it would succeed when the function does not exist and it would fail when the function already exists.